schemas/v1.0/schema.yaml

```diff
@@ -0,0 +1,563 @@
+      - workflowId
+      - steps
+    $ref: '#/$defs/specification-extensions'
+    unevaluatedProperties: false
+  step-object:
+    $comment: https://spec.openapis.org/arazzo/v1.0.0#step-object'
+    description: |-
+      Describes a single workflow step which MAY be a call to an
+      API operation (OpenAPI Operation Object or another Workflow Object)
+    type: object
+    properties:
+      stepId:
+        description: Unique string to represent the step
+        $dynamicAnchor: stepId
+        type: string
+        pattern: '^[A-Za-z0-9_\-]+$'
+      description:
+        description: A description of the step. CommonMark syntax MAY be used for rich text representation
+        type: string
+      operationId:
+        description: The name of an existing, resolvable operation, as defined with a unique operationId and existing within one of the sourceDescriptions
+        type: string
+      operationPath:
+        description: A reference to a Source combined with a JSON Pointer to reference an operation
+        type: string
+      workflowId:
+        description: The workflowId referencing an existing workflow within the Arazzo description
+        $dynamicRef: '#workflowId'
+      parameters:
+        description: A list of parameters that MUST be passed to an operation or workflow as referenced by operationId, operationPath, or workflowId
+        type: array
+        uniqueItems: true
+        minItems: 1
+        items: true
+      requestBody:
+        $ref: '#/$defs/request-body-object'
+      successCriteria:
+        description: A list of assertions to determine the success of the step
+        type: array
+        uniqueItems: true
+        minItems: 1
+        items:
+          $ref: '#/$defs/criterion-object'
+      onSuccess:
+        description: An array of success action objects that specify what to do upon step success
+        type: array
+        uniqueItems: true
+        items:
+          oneOf:
+            - $ref: '#/$defs/success-action-object'
+            - $ref: '#/$defs/reusable-object'
+      onFailure:
+        description: An array of failure action objects that specify what to do upon step failure
+        type: array
+        uniqueItems: true
+        items:
+          oneOf:
+            - $ref: '#/$defs/failure-action-object'
+            - $ref: '#/$defs/reusable-object'
+      outputs:
+        description: A map between a friendly name and a dynamic output value defined using a runtime expression
+        type: object
+        patternProperties:
+          '^[a-zA-Z0-9\.\-_]+$':
+            type: string
+    required:
+      - stepId
+    oneOf:
+      - required:
+          - operationId
+      - required:
+          - operationPath
+      - required:
+          - workflowId
+    allOf:
+      - if:
+          required:
+            - operationId
+          properties:
+            operationId: true
+        then:
+          properties:
+            parameters:
+              items:
```

Comment:
This should be a oneOf as the item could be a Parameter Object or a Reusable Object.

Reply:
Regarding this `allOf` definition related to `workflowId`, `operationId` and `operationPath`, the specification indicates there are conditions to be met.  That's the reason I created these conditional statements.

https://github.com/OAI/Arazzo-Specification/blob/df64e14d36e4c963256cbdf1b4f3759d8ccd95c5/versions/1.0.0.md?plain=1#L391

> The name location of the parameter. Possible values are "path", "query", "header", "cookie", or "body". When the step in context specifies a workflowId, then all parameters map to workflow inputs. In all other scenarios (e.g., a step specifies an operationId), the in field MUST be specified.

---

My interpretation of this section means if `operationId` or `operationPath` are the type defined, the reference can **only** be a Parameter object and the `in` attribute must be required.

If a `workflowId` is defined, either `parameterObject` or `reusableObject` may be defined.  

Is that a correction interpretation?